chalk-solve/src/clauses/builtin_traits/sized.rs

```diff
@@ -74,6 +74,15 @@ pub fn add_sized_program_clauses<I: Interner>(
             TypeName::Tuple(arity) => {
                 push_tuple_sized_conditions(db, builder, trait_ref, *arity, substitution)
             }
+            TypeName::Array => {
+                let interner = db.interner();
+                needs_impl_for_tys(
+                    db,
+                    builder,
+                    trait_ref,
+                    iter::once(substitution.at(interner, 0).assert_ty_ref(interner).clone()),
+                );
+            }
```

Comment:
Arrays are always `Sized`, proving `Sized` for the element type is a part of WF checks (which are not yet implemented (or designed, for that matter)), so here you should just `push_fact(trait_ref)`.